### ISSUE TITLE
multi @auth authorization Error (iam and groups)

### ISSUE THREAD

**Issue by hi120ki** [aws-amplify/amplify-cli#2277]
I write shema.graphql like this

```graphql
type Secret
  @model
  @auth(
    rules: [
      { allow: private, provider: iam, operations: [read] },
      { allow: groups, groups: ["Admin"] }
    ]
  ) {
  id: ID!
  secret: String!
}
```

Then, I can do listSecret from Lambda function that is authorized by IAM.
But, I can't do listSecret from Cognito User. ("Unauthorized" error occurred)

This graphql shema should be
- IAM : read
- Cognito User (group : Admin) : create, read, update, delete

But, there are no authority of Cognito User.

I'm waiting your great support.

**Comment by dabit3** [comment]
I think I'm running into either the same or a similar issue. Here is my info for reference, I hope this helps in solving the issue:

My base schema:

```
type Post @model
  @auth (
      rules: [
          { allow: groups, groups: ["Admin"] },
          { allow: public }
      ]
  ) {
  id: ID!
  title: String!
  content: String!
}
```

The generated GraphQL mutations look like this:

```
createPost(input: CreatePostInput!): Post
  @aws_api_key
updatePost(input: UpdatePostInput!): Post
  @aws_api_key
  @aws_cognito_user_pools
deletePost(input: DeletePostInput!): Post
  @aws_api_key
  @aws_cognito_user_pools
```

It seems that the `createPost` mutation is missing the `@aws_cognito_user_pools` directive.

**Comment by hi120ki** [comment]
I find that in my case, I have to write like this

```graphql
type Secret
  @model
  @auth(
    rules: [
      { allow: groups, groups: ["Admin"] },
      { allow: private },
      { allow: private, provider: iam, operations: [read] }
    ]
  ) {
  id: ID!
  secret: String!
}
```

This graphql shema's authority is

- Cognito User (group : Admin) : create, read, update, delete
- Cognito User : None
- IAM : read

and other examples,

```graphql
type Secret
  @model
  @auth(
    rules: [
      { allow: groups, groups: ["Admin"], operations: [create, update, delete] },
      { allow: private },
      { allow: private, provider: iam, operations: [read] }
    ]
  ) {
  id: ID!
  secret: String!
}
```

this graphql auth is

- Cognito User (group : Admin) : create, read, update, delete
- Cognito User : read
- IAM : read

In my case, the problem has been resolved.

And if you can, cloud you please make official documents more clearly?

Thank you.

**Comment by UnleashedMind** [comment]
We've already had a PR merged for this
https://github.com/aws-amplify/amplify-cli/pull/2305
and it's pending release.

**Comment by kaustavghosh06** [comment]
We released a new version of the CLI - v3.8.0 with a fix for this.

**Comment by dabit3** [comment]
Thanks @kaustavghosh06 !!
